### ISSUE TITLE
[ENHANCEMENT] Keeping numbering consistent when deleting items from a numbered list

### ISSUE THREAD

**Issue by peczony** [Zettlr/Zettlr#937]
## Description

There should be a way to keep numbering consistent when deleting items from a numbered list.
What I’m talking about is: suppose you have a list like

```
1. one
2. two
3. three
4. four
```

Then you delete point 2. You are left with

```
1. one
3. three
4. four
```

That’s not a problem if you export the content but when you e.g. copy Markdown, the inconsistency is still there. It would be better if there was some way to recalculate the numbering.

## Proposed Changes

I can think of two possible ways to solve this.

(1) When one presses enter to create a new list item, the numbering could be automatically recalculated, so instead of

```
1. one
3. three
4. four
5.
```

it would be

```
1. one
2. three
3. four
4.
```

Recalculating should also be done on indent change as it effectively creates new lists?

(2) A context menu item that would allow to recalculate numbering in the currently selected list

**Comment by sebastienvercammen** [comment]
Side note: GitHub supports the following:

```
1. a
1. b
1. c
```

and it'll render as

```
1. a
2. b
3. c
```

as demonstrated:
1. a
1. b
1. c

A similar approach will simplify all numbering. I don't know how this affects other systems though (but if it does, perhaps it's time for this to become a standard?).

**Comment by nathanlesage** [comment]
> A similar approach will simplify all numbering. I don't know how this affects other systems though (but if it does, perhaps it's time for this to become a standard?).

In Zettlr, you can also do lists like such:

```
#. One
#. Two
#. Three
```

And it'll automatically number; because this is a feature of Pandoc. Oh, and we might want to support that in the list indentation as well … let's remember this!

**Comment by peczony** [comment]
Sure, I’m totally fine with `1. 1.` or `#. #.` syntax. On the other hand it would be convenient to quickly see what the resulting numbering would be without first exporting the content. Like some hotkey for quick non-editable preview maybe?

**Comment by nathanlesage** [comment]
Of course, but there's no hotkey necessary. Just some improvements to the underlying algorithm, which is currently based on a CodeMirror plugin. We'd need to rewrite that thing. Possible, but quite some work!

**Comment by peczony** [comment]
I mean maybe “show quick html preview instead of editable text” is less work than rewriting the plugin, with the same effect (and maybe even better for some use cases) :)

**Comment by nathanlesage** [comment]
> I mean maybe “show quick html preview instead of editable text” is less work than rewriting the plugin

Actually it's the opposite :D At least when it comes to sane code and my own mental health! ;D

**Comment by finzzz** [comment]
Don't close this. I keep encountering this bug very often.

**Comment by nathanlesage** [comment]
It's not a bug, it's simply that the algorithm is not as smart as it could be ;)

**Comment by finzzz** [comment]
Can I turn off this feature?

**Comment by nathanlesage** [comment]
Nope

**Comment by braykov** [comment]
I would just like to be able to set an option for the numbered list to not at all increment the numbers, but just put 1., 1., 1.
This would solve all problems from above. It is also the standard accepted by GitLab.
It is also better for when you view diffs in version control.

**Comment by nathanlesage** [comment]
@braykov Mh, in the Pandoc-context the equivalent would be `#.`. I'm not sure if the editor accepts this syntax, though …

**Comment by braykov** [comment]
@nathanlesage unfortunately that is not standard and not supported by GitLab.

**Comment by MediocrityXT** [comment]
Need this EHANCEMENT!!! The long numbering list makes me really annoyed when lots of numbering need rewriting.
[recent numbering fixing dates back to 2018...](https://github.com/codemirror/CodeMirror/pull/5318)
After you press enter and generate a new list item, rewrite all items' numbering after the new item.I guess it's not that hard but don't know why nobody fix it

**Comment by nathanlesage** [comment]
@MediocrityXT The PR you referenced was about continuing the correct numbering, not when you delete an item. And yes, it is hard to fix it and that's why it has not yet been fixed.

**Comment by yin-ori** [comment]
May I also add to the question that whenever I try to do something like
1. first point
2. second point 
3. third point

and then have sub-points under it like 
1. first point
    *  sub
3. second point

it will leave the list as 1., 3. and skipping 2 - that is really a bummer as it makes numbered lists unusable for me.
